### PR TITLE
Spring cleaning

### PR DESCRIPTION
**Pull request by tie-rack**
* YourKit!
* Newrelic!
* Uberjar!
* Dependency version updates!
* Cleanup!

### CODE REVIEW

File: resources/leiningen/new/kraken_http_api/PROJECT@.service.template [hunk 1]
```diff
@@ -25,12 +21,17 @@ ExecStartPre=/bin/bash -c 'sleep 2 && curl -s http://${COREOS_PRIVATE_IPV4}:8500
 ExecStartPre=/usr/bin/docker pull ${DOCKER_REPO}:${VERSION}
 
 ExecStart=/bin/bash -c 'docker run --name ${CONTAINER} --restart=always \
-  --link rabbitmq:rabbitmq \
-  --link wildfly:wildfly \
+  --dns $COREOS_PRIVATE_IPV4 \
   --env ALLOWED_ORIGINS="$(curl -s http://${COREOS_PRIVATE_IPV4}:8500/v1/kv/{{name}}/allowed-origins?raw)" \
+  --env NEW_RELIC_LICENSE_KEY=$(curl -s http://${COREOS_PRIVATE_IPV4}:8500/v1/kv/new-relic/license-key?raw) \
+  --env RABBITMQ_PORT_5672_TCP_ADDR=$(curl -s http://${COREOS_PRIVATE_IPV4}:8500/v1/kv/rabbitmq/host?raw) \
```
Comment (cap10morgan) [inline resources/leiningen/new/kraken_http_api/PROJECT@.service.template:27]: Hmm, yeah, now that I think about it, we probably want this in the `-works` template too.

File: resources/leiningen/new/kraken_http_api/docker-compose.yml [hunk 2]
```diff
@@ -2,19 +2,10 @@ app:
...
-    - rabbitmq
   ports:
-    - "59990:9990"
     - "58080:8080"
   environment:
-    ADMIN_USERNAME: admin
-    ADMIN_PASSWORD: admin
+    ALLOWED_ORIGINS: '[".*"]'
 rabbitmq:
   image: rabbitmq:3.5.3-management
```
Comment (cap10morgan) [inline resources/leiningen/new/kraken_http_api/docker-compose.yml]: This should be bumped up to `rabbitmq:3.6.9-management` now.

File: resources/leiningen/new/kraken_http_api/server.clj [hunk 3]
```diff
@@ -19,5 +19,5 @@
 
 (defn -main [& args]
   (let [rabbit-resources (queue/initialize)]
-    (start-http-server (config [:server]))
+    (start-http-server)
```
Comment (cap10morgan) [inline resources/leiningen/new/kraken_http_api/server.clj]: What's the motivation for this change?

**Review by cap10morgan — CHANGES_REQUESTED** [review]

File: resources/leiningen/new/kraken_http_api/server.clj (hunk 3 above)
Comment (tie-rack) [reply resources/leiningen/new/kraken_http_api/server.clj]: I was using user-http-api as my guide and it doesn't pass that arg. All the rest do, though. I'll revert that.

File: resources/leiningen/new/kraken_http_api/Dockerfile [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
 
 COPY . /usr/src/{{name}}
 
-RUN lein test
-RUN lein immutant war --name {{name}} --destination target --nrepl-port=11111 --nrepl-start --nrepl-host=0.0.0.0
+RUN lein with-profiles $env,test test
+RUN lein with-profile $env uberjar
+
+CMD java ${JVM_OPTS:--XX:+UseG1GC} \
+    -javaagent:resources/jars/com.newrelic.agent.java/newrelic-agent.jar \
```
Comment (cap10morgan) [inline resources/leiningen/new/kraken_http_api/Dockerfile]: Maybe we should get the cgroup heap stuff in here too? And in the works template?

File: resources/leiningen/new/kraken_http_api/PROJECT@.service.template (hunk 1 above)
Comment (tie-rack) [reply resources/leiningen/new/kraken_http_api/PROJECT@.service.template:27]: I'll open a new PR for the -works template today.

File: resources/leiningen/new/kraken_http_api/Dockerfile (hunk 4 above)
Comment (tie-rack) [reply resources/leiningen/new/kraken_http_api/Dockerfile]: Yep. Update soon.

**Review by cap10morgan — APPROVED** [review]
🐞

**Pull request merged**